crates/sui-core/src/authority/authority_store.rs

```diff
@@ -713,61 +713,21 @@ impl AuthorityStore {
             iter::once((transaction_digest, certificate.serializable_ref())),
         )?;
 
-        self.sequence_tx(
-            write_batch,
-            inner_temporary_store,
-            transaction_digest,
-            effects,
-            effects_digest,
-        )
-        .await?;
-
-        self.effects_notify_read.notify(transaction_digest, effects);
-
-        Ok(())
-    }
-
-    /// Persist temporary storage to DB for genesis modules
-    pub async fn update_objects_state_for_genesis(
-        &self,
-        inner_temporary_store: InnerTemporaryStore,
-        transaction_digest: TransactionDigest,
-    ) -> Result<(), SuiError> {
-        debug_assert_eq!(transaction_digest, TransactionDigest::genesis());
-        let write_batch = self.perpetual_tables.certificates.batch();
-        self.batch_update_objects(
-            write_batch,
-            inner_temporary_store,
-            transaction_digest,
-            UpdateType::Genesis,
-        )
-        .await?;
-        Ok(())
-    }
-
-    async fn sequence_tx(
-        &self,
-        write_batch: DBBatch,
-        inner_temporary_store: InnerTemporaryStore,
-        transaction_digest: &TransactionDigest,
-        effects: &SignedTransactionEffects,
-        effects_digest: &TransactionEffectsDigest,
-    ) -> SuiResult {
-        // Safe to unwrap since UpdateType::Transaction ensures we get a sequence number back.
-        self.batch_update_objects(
-            write_batch,
-            inner_temporary_store,
-            *transaction_digest,
-            UpdateType::Transaction(*effects_digest),
-        )
-        .await?;
+        // Add batched writes for objects and locks.
+        write_batch = self
+            .update_objects_and_locks(
+                write_batch,
+                inner_temporary_store,
+                *transaction_digest,
+                UpdateType::Transaction(*effects_digest),
+            )
+            .await?;
 
         // Store the signed effects of the transaction
         // We can't write this until after sequencing succeeds (which happens in
         // batch_update_objects), as effects_exists is used as a check in many places
         // for "did the tx finish".
-        let batch = self.perpetual_tables.executed_effects.batch();
```

Comment:
oh this was my bad. i missed this when removing lock service. Dang!

Reply:
Yeah it is easy to miss that different batches are used ..